dialogue-futures/src/test/java/com/palantir/dialogue/futures/DialogueFuturesTest.java

```diff
@@ -235,6 +237,36 @@ void testCatchingAllAsync_resultCancel_completed(Transformer transformer) throws
         assertThat(transformed.get()).isEqualTo("b");
     }
 
+    @Test
+    void testSafeDirectExecutor() {
+        AtomicInteger task = new AtomicInteger();
+        DialogueFutures.safeDirectExecutor().execute(task::incrementAndGet);
+        assertThat(task).hasValue(1);
+    }
+
+    @Test
+    void testSafeDirectExecutorDoesNotThrowError() {
+        Executor executor = DialogueFutures.safeDirectExecutor();
+        assertThatCode(() -> executor.execute(() -> {
+                    throw new Error();
+                }))
+                .doesNotThrowAnyException();
+    }
+
+    @Test
+    void testAddDirectListenerContinuesAfterError() {
+        SettableFuture<String> future = SettableFuture.create();
+        AtomicInteger invocations = new AtomicInteger();
+        Runnable listener = () -> {
+            invocations.incrementAndGet();
+            throw new Error();
+        };
+        DialogueFutures.addDirectListener(future, listener);
+        DialogueFutures.addDirectListener(future, listener);
+        future.set("value");
+        assertThat(invocations).hasValue(2);
+    }
```

Comment:
This one is :hot_pepper: without the fix. `future.set` throws the Error and invocations value is 1.